PyInputToolExample/engine.py

```diff
@@ -0,0 +1,210 @@
+import AlteryxPythonSDK
+import xml.etree.ElementTree as ET
+import csv
+import os
+
+class AyxPlugin:
+    def __init__(self, n_tool_id, engine_interface, generic_engine, output_anchor_mgr):
+        # initialize *all* members that will be used (for PEP8 compliance)
+
+        # miscellaneous variables
+        self.n_tool_id = n_tool_id
+        self.name = str('PyInputTool_') + str(self.n_tool_id)
+        self.closed = False
+        self.initialized = False
+
+        # engine handles
+        self.alteryx_engine = engine_interface
+        self.generic_engine = generic_engine
+
+        # output anchor management
+        self.output_anchor_mgr = output_anchor_mgr
+        self.output_anchor = None
+
+        # record management
+        self.record_info_out = None
+        self.record_creator = None
+        self.output_field = None
+
+        #
+        # create your custom members here and give them default values
+        #
+        self.file_input_name = 'C:\\Users\\username\\Desktop\\PythonInputTest.csv'
+        self.file_out = 'file object'
+        self.file_reader = 'mapped file object'
```

Comment:
I would argue everything should be declared in `init`. I shared the rationale some in this other comment: https://github.com/alteryx/python-sdk-samples/commit/ef76ab73f750c703eb373e390e578017290f262c#commitcomment-23214041

As for what they should be declared as, if there's not a better default, I use `None`.